### PR TITLE
Feature: Focus existing tab when opening location from external process

### LINKED ISSUE #6925: Feature: Focus existing tab instead of creating duplicate when opening from external process
### Description

Opening a folder from an external link (e.g. download from a browser) will open a second tab of the same folder instead of re-using the first instance of the tab.  This occurs with the download folder at a minimum

### Steps To Reproduce

1. Download file from Chrome.
2. Click to open downloads folder from Chrome
3. New Downloads folder tab is opened in Files.
4. Repeating steps 1-3, creates a new Downloads tab instead of re-using and activating the existing Downloads tab already in Files.

### Expected behavior

Expected Downloads folder tab to be used instead of a second tab for the same folder being opened.

### Files Version

Version: 2.0.34.0 

### Windows Version

OS Version: 10.0.22000.318

### Relevant Assets

_No response_

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #6925

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Set Files as default file manager
2. Confirmed opening path from external location focuses on existing tabs
3. Confirmed opening path from external location adds a new tab if needed

### CODE REVIEW

File: src/Files.App/MainWindow.xaml.cs [hunk 1]
```diff
@@ -286,7 +286,16 @@ async Task PerformNavigationAsync(string payload, string selectItem = null)
 					// Bring to foreground (#14730)
 					Win32Helper.BringToForegroundEx(new(WindowHandle));
 
-					await NavigationHelpers.AddNewTabByParamAsync(typeof(ShellPanesPage), paneNavigationArgs);
+					// Check if there is an existing tab with this path
+					var existingTab = MainPageViewModel.AppInstances
+						.Select((tabItem, idx) => new { tabItem, idx })
+						.FirstOrDefault(x => x.tabItem.ToolTipText == paneNavigationArgs.LeftPaneNavPathParam ||
+						 x.tabItem.ToolTipText == paneNavigationArgs.RightPaneNavPathParam)?.idx ?? -1;
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/MainWindow.xaml.cs:293]: tabItem.NavigationParameter.NavigationParameter is PaneNavigationArguments paneNavigationArguments could work

[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/MainWindow.xaml.cs:293]: I'm surprised it doesn't have Path prop, it definitely should have it.

[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/MainWindow.xaml.cs:293]: Isnt that actually pane args instance?

File: src/Files.App/MainWindow.xaml.cs [hunk 2]
```diff
@@ -286,7 +286,16 @@ async Task PerformNavigationAsync(string payload, string selectItem = null)
 					// Bring to foreground (#14730)
 					Win32Helper.BringToForegroundEx(new(WindowHandle));
 
-					await NavigationHelpers.AddNewTabByParamAsync(typeof(ShellPanesPage), paneNavigationArgs);
+					var existingTabIndex = MainPageViewModel.AppInstances
+						.Select((tabItem, idx) => new { tabItem, idx })
+						.FirstOrDefault(x => x.tabItem.NavigationParameter.NavigationParameter is PaneNavigationArguments paneArgs &&
+							(paneNavigationArgs.LeftPaneNavPathParam == paneArgs.LeftPaneNavPathParam || 
+							paneNavigationArgs.LeftPaneNavPathParam == paneArgs.RightPaneNavPathParam))?.idx ?? -1;
```
Comment (0x5bfa) [inline src/Files.App/MainWindow.xaml.cs:293]:
```suggestion
						.FirstOrDefault(x =>
							x.tabItem.NavigationParameter.NavigationParameter is PaneNavigationArguments paneArgs &&
							(paneNavigationArgs.LeftPaneNavPathParam == paneArgs.LeftPaneNavPathParam || 
							paneNavigationArgs.LeftPaneNavPathParam == paneArgs.RightPaneNavPathParam))?.idx ?? -1;
```

**Review by 0x5bfa — APPROVED** [review]
LGTM, code wise.

**Comment by yaira2** [comment]
@0x5bfa did you also check functionality wise?

**Comment by 0x5bfa** [comment]
I'll do that in a day I think.

**Pull request merged**